crates/iota-indexer/src/apis/governance_api.rs

```diff
@@ -450,32 +461,63 @@ impl<T: R2D2Connection + 'static> GovernanceReadApi<T> {
                     false,
                 )
             })
-            .collect::<Vec<(IotaAddress, ObjectID, ObjectID, u64, bool)>>();
+            .collect::<Vec<ValidatorTable>>();
 
-        // Get inactive validator rate tables
-        for df in self
-            .inner
-            .get_dynamic_fields_in_blocking_task(
+        self.validator_exchange_rates(tables).await
+    }
+
+    /// Check for validators in the `Candidate` state and get its exchange rate.
+    /// For these validators, their exchange rates should not be cached as
+    /// their state can occur during an epoch or across multiple ones. In
+    /// contrast, exchange rates for `Active` and `Inactive` validators can
+    /// be cached, as their state changes only at epoch change.
+    pub async fn candidate_validators_exchange_rate(
+        &self,
+        system_state_summary: &IotaSystemStateSummary,
+    ) -> Result<Vec<ValidatorExchangeRates>, IndexerError> {
+        let tables = self
+            .validator_summary_from_system_state(
                 system_state_summary.validator_candidates_id,
-                None,
-                system_state_summary.validator_candidates_size as usize,
+                system_state_summary.validator_candidates_size,
+                |df| {
+                    bcs::from_bytes::<IotaAddress>(&df.bcs_name).map_err(|err| {
+                        IotaError::ObjectDeserialization {
+                            error: err.to_string(),
+                        }
+                        .into()
```

Comment:
Can be simplified with `map_err(IndexerError::from)` (there is a `From<bcs::Error>` implementation)

Reply:
[Fixed](https://github.com/iotaledger/iota/pull/4313/commits/05a02543dfc4b99f24c0ce410aaffc9139e5bf30#diff-324f865071b593cb4ed59dd865b9600fae68065a818e622b14f4f5554812fd98R474)